app/javascript/src/components/Dashboard/Notes/NewNoteForm.jsx

```diff
@@ -1,36 +1,107 @@
+      ? moment(new Date(values.dueDate), "DD/MM/YYYY").format("MMMM DD, YYYY")
+      : "";
+    addNewNote(values);
+    onClose();
   };
   return (
     <Formik
       initialValues={{
         title: "",
+        tag: "",
         description: "",
+        contact: "",
+        isDueDate: false,
+        dueDate: "",
       }}
       onSubmit={handleSubmit}
       validationSchema={yup.object({
         title: yup.string().required("Title is required"),
+        tag: yup.string().required("Tag is required"),
         description: yup.string().required("Description is required"),
+        contact: yup.string().required("Contact is required"),
+        isDueDate: yup.boolean(),
+        dueDate: yup.string().when("isDueDate", {
+          is: true,
+          then: yup
+            .string()
+            .required("Due Date is required")
+            .validDateInput("Enter a valid date in DD/MM/YYYY format only"),
+          otherwise: yup.string(),
+        }),
       })}
     >
-      {({ isSubmitting }) => (
+      {({ isSubmitting, handleChange, setFieldValue, values }) => (
         <Form>
-          <Input label="Title" name="title" className="mb-6" />
-          <Textarea label="Description" name="description" rows={8} />
+          <Input label="Note Title" name="title" className="mb-6" />
+          <Select
+            label="Tags"
+            name="tag"
+            placeholder="Select a tag"
+            className="mb-6"
+            onChange={option => {
+              handleChange(option.value);
+              setFieldValue("tag", option.value);
+            }}
+            options={[
+              { value: "Internal", label: "Internal" },
+              { value: "Agile Workflow", label: "Agile Workflow" },
+              { value: "Bug", label: "Bug" },
+            ]}
+          />
+          <Textarea
+            label="Note Description"
+            name="description"
+            rows={4}
+            className="mb-6"
+          />
+          <Select
+            label="Assigned Contact"
+            name="contact"
+            placeholder="Select a contact"
+            className="mb-6"
+            onChange={option => {
+              handleChange(option.value);
+              setFieldValue("contact", option.value);
+            }}
+            options={[
+              { value: "Neeraj Singh", label: "Neeraj Singh" },
+              { value: "Vinay V", label: "Vinay V" },
+              { value: "Charlie Smith", label: "Charlie Smith" },
+              { value: "Karthik Menon", label: "Karthik Menon" },
+            ]}
+          />
+          <div className="flex flex-row items-center justify-between mb-6">
+            Add Due Date to Note
+            <Switch name="isDueDate" />
+          </div>
+          {values.isDueDate && (
+            <Input
+              placeholder="Ex: 21/10/2021"
+              label="Due Date"
+              name="dueDate"
+            />
+          )}
```

Comment:
Rather than using `mb-6` in individual blocks, please use `space-x-6` in the parent container. 

Please refer [space between](https://tailwindcss.com/docs/space#class-reference) for more details.